test/model/content_test.dart

```diff
@@ -521,6 +521,118 @@ class ContentExample {
+    '</p>\n'
+    '<div class="youtube-video message_inline_image">'
+      '<a data-id="aqz-KE-bpKQ" href="https://www.youtube.com/watch?v=aqz-KE-bpKQ">'
+        '<img src="/external_content/ecb96e8f884f481c4bc0179287a44ab9014aa78f/68747470733a2f2f692e7974696d672e636f6d2f76692f61717a2d4b452d62704b512f64656661756c742e6a7067"></a></div>', [
+    ParagraphNode(links: null, nodes: [
+      LinkNode(url: 'https://www.youtube.com/watch?v=aqz-KE-bpKQ', nodes: [TextNode('https://www.youtube.com/watch?v=aqz-KE-bpKQ')]),
+    ]),
+    VideoNode(
+      srcUrl: "https://www.youtube.com/watch?v=aqz-KE-bpKQ",
+      previewImageUrl: "/external_content/ecb96e8f884f481c4bc0179287a44ab9014aa78f/68747470733a2f2f692e7974696d672e636f6d2f76692f61717a2d4b452d62704b512f64656661756c742e6a7067"
+    ),
+  ]);
+
+  static const videoEmbedYoutubeClassNameReordered = ContentExample(
+    'video preview for youtube embed with thumbnail, (hypothetical) class name reorder',
+    null,
+    '<p>'
+      '<a href="https://www.youtube.com/watch?v=aqz-KE-bpKQ">https://www.youtube.com/watch?v=aqz-KE-bpKQ</a>'
+    '</p>\n'
+    '<div class="message_inline_image youtube-video">'
+      '<a data-id="aqz-KE-bpKQ" href="https://www.youtube.com/watch?v=aqz-KE-bpKQ">'
+        '<img src="/external_content/ecb96e8f884f481c4bc0179287a44ab9014aa78f/68747470733a2f2f692e7974696d672e636f6d2f76692f61717a2d4b452d62704b512f64656661756c742e6a7067"></a></div>', [
+    ParagraphNode(links: null, nodes: [
+      LinkNode(url: 'https://www.youtube.com/watch?v=aqz-KE-bpKQ', nodes: [TextNode('https://www.youtube.com/watch?v=aqz-KE-bpKQ')]),
+    ]),
+    VideoNode(
+      srcUrl: "https://www.youtube.com/watch?v=aqz-KE-bpKQ",
+      previewImageUrl: "/external_content/ecb96e8f884f481c4bc0179287a44ab9014aa78f/68747470733a2f2f692e7974696d672e636f6d2f76692f61717a2d4b452d62704b512f64656661756c742e6a7067"
+    ),
+  ]);
+
+  static const videoEmbedVimeoPreviewDisabled = ContentExample(
+    'video preview for vimeo embed with realm link previews disabled',
+    null,
+    '<p>'
+      '<a href="https://vimeo.com/1084537">https://vimeo.com/1084537</a></p>', [
+    ParagraphNode(links: null, nodes: [
+      LinkNode(url: 'https://vimeo.com/1084537', nodes: [TextNode('https://vimeo.com/1084537')]),
+    ]),
+  ]);
+
+  static const videoEmbedVimeoPreviewEnabled = ContentExample(
+    'video preview for vimeo embed with realm link previews enabled',
+    null,
+    '<p>'
+      '<a href="https://vimeo.com/1084537">Vimeo - Big Buck Bunny</a>'
+    '</p>\n'
+    '<div class="embed-video message_inline_image">'
+      '<a data-id="&lt;iframe src=&quot;https://player.vimeo.com/video/1084537?app_id=122963&quot; width=&quot;640&quot; height=&quot;360&quot; frameborder=&quot;0&quot; allow=&quot;autoplay; fullscreen; picture-in-picture; clipboard-write&quot; title=&quot;Big Buck Bunny&quot;&gt;&lt;/iframe&gt;" href="https://vimeo.com/1084537" title="Big Buck Bunny">'
```

Comment:
The value of the `data-id` attribute is a blob of HTML? For an iframe element? Wacky. That's not much of an "ID".

If the Zulip server produces this, or has produced it in the past, then it's good to have in a test case. But it smells rather like a bug. It'd be good to report it in a chat thread in `#issues` and see what people say.

This has enough resemblance to a potential HTML injection that I waffled on whether to treat it as a potential security issue (which should be sent to security@zulip.com rather than discussed in public). So I poked around in the server implementation, and the logic for this is pretty explicit in `zerver/lib/markdown/__init__.py`:
```python
        elif extracted_data.type == "video":
            self.add_a(
                # …
                class_attr="embed-video message_inline_image",
                data_id=extracted_data.html,
```

That makes it look intentional enough that I'm OK with keeping the discussion public. But I'd still be interested in what the story is with calling this "id" when it's actually some HTML for an iframe.

Reply:
[Reported](https://chat.zulip.org/#narrow/stream/9-issues/topic/Vimeo.20link.20previews.20HTML.20.22data-id.22.20isn't.20a.20.20Vimeo.20video.20ID/near/1767563)